src/wms/src/main/java/org/geoserver/wms/WMSRequests.java

```diff
@@ -309,6 +311,7 @@ private static String getLayerStyles(Layer[] layers) {
                     }
                 }
             }
+            index = getRawLayerIndex(req, index);
```

Comment:
Thinking out loud, would it be better to have the index be -1 by default, and have it throw an exception if the layer is not found?

Reply:
I'm not sure about the -1 but an exception is now thrown if it can't find the correct index.